internal/handlers/pg/pgdb/query.go

```diff
@@ -60,7 +60,7 @@ func Explain(ctx context.Context, tx pgx.Tx, qp *QueryParams) (*types.Document,
 		return nil, lazyerrors.Error(ErrTableNotExist)
 	}
 
-	table, err := getMetadata(ctx, tx, qp.DB, qp.Collection)
+	table, err := newMetadata(tx, qp.DB, qp.Collection).getTableName(ctx)
```

Comment:
At the moment for pushdown check, we check if [`Filter` exists](https://github.com/FerretDB/FerretDB/blob/main/internal/handlers/pg/msg_explain.go#L88), but now we use index condition so it looks like those cases it uses `Index Cond` for explain plan. See the doc https://www.postgresql.org/docs/current/using-explain.html

debug output to show what is available in the plan:
![image](https://user-images.githubusercontent.com/24355139/223311996-23841338-1e10-4246-9c71-d6f5b76d05c4.png)

So to check pushdown is done, it is when any of `Filter`, `Index Cond` or any other index is present in explain plan.

Reply:
Thanks a lot!